### PR TITLE
Add logic to write BNC data to csv & improve previously written code

### PR DESCRIPTION
**Pull request by ThePhilgrim**

### CODE REVIEW

**Comment by ThePhilgrim** [comment]
TODO:
Screenshot 2021-05-19 at 00 29 18

File: ntrans_dataprep.py [hunk 1]
```diff
@@ -17,36 +18,37 @@
...
-    pass
 
+    filenames = ["2-grams", "3-grams", "4-grams", "5-grams", "6-grams"]
+    pathlib.Path("ngrams").mkdir(exist_ok=True)
 
-def count_ngram_frequency(twograms, threegrams, fourgrams, fivegrams, sixgrams):
+    for n, counter in n_to_ngrams.items():
+
+        # "n - 2" to get index of "filenames"
+        file_path = "./ngrams/" + filenames[n - 2] + ".csv"
```
Comment (Akuli) [inline ntrans_dataprep.py]:
```suggestion
        file_path = f"./ngrams/{n}-grams.csv"
```

And you can delete `filenames` variable.

File: ntrans_dataprep.py [hunk 2]
```diff
@@ -17,36 +18,37 @@
...
+
+        with open(file_path, mode="w") as write_data_file:
+            data_writer = csv.writer(write_data_file, delimiter=",")
```
Comment (Akuli) [inline ntrans_dataprep.py]: Comma is default:

```suggestion
            data_writer = csv.writer(write_data_file)
```

File: ntrans_dataprep.py [hunk 3]
```diff
@@ -17,36 +18,37 @@
...
-        counted_four_grams,
-        counted_five_grams,
-        counted_six_grams,
-    )
+
+    for n in n_to_ngrams.keys():
+        n_to_ngrams[n] = collections.Counter(n_to_ngrams[n])
+
+    # print(n_to_ngrams[2])
+    write_data_to_csv(n_to_ngrams)
```
Comment (Akuli) [inline ntrans_dataprep.py]: This is what I meant yesterday:

```suggestion
    write_data_to_csv({
        n: collections.Counter(ngrams)
        for n, ngrams in n_to_ngrams.items()
    })
```

**Review by Akuli — COMMENTED** [review]
Looks good overall. Small nits.

File: ntrans_dataprep.py (hunk 1 above)
Comment (Akuli) [reply ntrans_dataprep.py]: There's also a comment above this line of code. That can be deleted too.

File: ntrans_dataprep.py [hunk 4]
```diff
@@ -17,36 +18,35 @@
...
-    counted_five_grams,
-    counted_six_grams,
-):
+def write_data_to_csv(n_to_ngrams):
     """
     Writes the N-grams to CSV files. These CSV files are read from ntrans.py
     """
-    pass
 
+    filenames = ["2-grams", "3-grams", "4-grams", "5-grams", "6-grams"]
```
Comment (Akuli) [inline ntrans_dataprep.py]:
```suggestion
```

File: ntrans_dataprep.py [hunk 5]
```diff
@@ -17,36 +18,35 @@
...
+    pathlib.Path("ngrams").mkdir(exist_ok=True)
 
-def count_ngram_frequency(twograms, threegrams, fourgrams, fivegrams, sixgrams):
+    for n, counter in n_to_ngrams.items():
+
+        # "n - 2" to get index of "filenames"
```
Comment (Akuli) [inline ntrans_dataprep.py]:
```suggestion
```

**Review by Akuli — COMMENTED** [review]
Some leftovers from `file_path` cleanup:

**Review by Akuli — APPROVED** [review]

**Pull request merged**